frontend/src/components/pages/GDCDataEntry.tsx

```diff
@@ -0,0 +1,953 @@
+                      isInvalid={errorGoalMunicipality}
+                      errorBorderColor="crimson"
+                    >
+                      {municipalities &&
+                        municipalities.map((muni) => (
+                          <option key={muni.code} value={muni.code}>
+                            {muni.name}
+                          </option>
+                        ))}
+                    </Select>
+                  </FormControl>
+                  <FormControl id="goal-file" isRequired>
+                    <FormLabel>File:</FormLabel>
+                    <Input
+                      errorBorderColor="crimson"
+                      type="file"
+                      style={{
+                        paddingTop: '0.25rem',
+                        paddingLeft: '0.25rem',
+                        height: '40px',
+                        opacity: 0,
+                        zIndex: 20,
+                        cursor: 'pointer',
+                      }}
+                      onChange={(evt) => setGoalFile(evt.target.files ? evt.target.files[0] : null)}
+                      accept="text/csv, .csv"
+                    />
+                    <InputGroup
+                      style={{
+                        height: '40px',
+                        marginTop: '-40px',
+                      }}
+                    >
+                      <InputLeftElement
+                        style={{
+                          width: '6rem',
+                          cursor: 'pointer',
+                        }}
+                      >
+                        <Button>Browse...</Button>
+                      </InputLeftElement>
+                      <Input
+                        value={goalFile ? goalFile.name : ''}
+                        isReadOnly
+                        style={{
+                          paddingLeft: '7rem',
+                          cursor: 'pointer',
+                        }}
+                        isInvalid={errorGoalFile}
+                        errorBorderColor="crimson"
+                      />
+                    </InputGroup>
+                  </FormControl>
+                  <Spacer m="0.5rem" />
+                  <FormControl id="goal-dummy">
+                    <FormLabel>Are these dummy goals?</FormLabel>
+                    <Checkbox
+                      isChecked={goalDummy}
+                      onChange={(evt) => setGoalDummy(evt.currentTarget.checked)}
+                    >
+                      Dummy goals
+                    </Checkbox>
+                  </FormControl>
+                  <Spacer m="2rem" />
+                  <Button onClick={onSubmitGoals}>Upload goals</Button>
+                  <Spacer m="2rem" />
+                  <CSVGoalInfo />
+                </Stack>
+              </TabPanel>
+            </TabPanels>
+          </Tabs>
+        </Stack>
+      </Flex>
+    </Stack>
+  );
+};
+
+export default GDCDataEntry;
```

Comment:
This file is a bit too long, might consider refactoring it in the future 😅

Reply:
The length is mostly due to the JSX for the CSV requirements, but I agree, should be refactored a bit in the future.